File: algorithms/implementation/cats-and-a-mouse/cats-and-a-mouse-python/tests/test_catAndMouse.py
from catAndMouse import catAndMouse

class TesteCatAndMouse:
    
    def testeCaso1(self):
        
        x = 1
        y = 2
        z = 3

        esperado = 'Cat B'

        resultado = catAndMouse(x, y, z)
        
        assert resultado == esperado
        
    def testeCaso2(self):
        
        x = 1
        y = 3
        z = 2

        esperado = 'Mouse C'

        resultado = catAndMouse(x, y, z)
        
        assert resultado == esperado
        
        
    def testeCaso3(self):
        
        x = 33
        y = 86
        z = 59

        esperado = 'Cat A'

        resultado = catAndMouse(x, y, z)
        
        assert resultado == esperado

File: algorithms/implementation/cats-and-a-mouse/cats-and-a-mouse-javascript/js/test_catAndMouse.js
import {catAndMouse} from '../js/catAndMouse.js';
import assert from 'assert';

describe('catAndMouse', function() {

    it('Deve retornar quando estiver mais próximo do rato: Cat B', function() {

        let x = '1';
        let y = '2';
        let z = '3';

        let esperado = 'Cat B';

        let resultado = catAndMouse(x, y, z)

        assert.strictEqual(resultado, esperado);
    });
});

describe('catAndMouse', function() {

    it('Deve retornar quando estiver mais próximo do rato: Mouse C Escape', function() {

        let x = '1';
        let y = '3';
        let z = '2';

        let esperado = 'Mouse C';

        let resultado = catAndMouse(x, y, z)

        assert.strictEqual(resultado, esperado);
    });
});

describe('catAndMouse', function() {

    it('Deve retornar quando estiver mais próximo do rato: Cat A', function() {

        let x = '33';
        let y = '86';
        let z = '59';

        let esperado = 'Cat A';

        let resultado = catAndMouse(x, y, z)

        assert.strictEqual(resultado, esperado);
    });
});